### DISCUSSION TITLE
Add a way to get item by one of its unique attributes

### DISCUSSION THREAD

**Discussion by garretcharp** [sensedeep/dynamodb-onetable#142] [Ideas]
I think it would be a cool addition to be able to get an item by one of its unique attributes, without setting it on the top-level item by secondary indexes. You could handle this by saving the hash/sort key that "owns" the unique value on the unique item.

For example I have a user item with the following schema:
```js
const User = {
	pk: {
		type: String,
		value: '${_type}:${id}'
	},
	sk: {
		type: String,
		value: '${_type}:'
	},

	gsi1pk: {
		type: String,
		value: '${_type}:email:${email}'
	},
	gsi1sk: {
		type: String,
		value: '${_type}:email:'
	},

	gsi2pk: {
		type: String,
		value: '${_type}:handle:${handle}'
	},
	gsi2sk: {
		type: String,
		value: '${_type}:handle:'
	},

	id: {
		type: String,
		uuid: true
	},
	handle: {
		type: String,
		required: true,
		unique: true
	},
	email: {
		type: String,
		required: true,
		unique: true
	},
}
```

A cool feature would allow me to remove gsi1 and gsi2 values from that schema and the unique model handled by the library would be able to setup the query for me. On the underline unique item that gets saved you would add 2 attributes "itempk" and "itemsk" (or whatever the index values for hash/sort might be prefixed by item). Then whenever I want to look up the original item by a unique value the library would retrieve the unique value item and then query the model by its primary index.

**Comment by mobsense** [comment]
Hi Garret ... good suggestion. The tradeoff is the cost of the extra attributes in the unique item vs the utility of being able to fetch by unique attribute.

Do any others have opinions?
